src/main/java/com/sap/charging/server/api/v1/OptimizeChargingProfilesController.java

```diff
@@ -24,61 +25,59 @@
 
-	@Override
-	public int getVerbosity() {
-		return Simulation.verbosity;
-	}
+    @Override
+    public int getVerbosity() {
+        return Simulation.verbosity;
+    }
 
-	@CrossOrigin(origins = "*")	
-	@ApiOperation(value = "Optimize Charging Profiles")
-    @PostMapping(path="/api/v1/OptimizeChargingProfiles", produces=MediaType.APPLICATION_JSON_VALUE)
-    @ResponseBody
-    public ResponseEntity<Object> optimizeChargingProfiles( @ApiParam @RequestBody OptimizeChargingProfilesRequest request) {
-    	
-    	Simulation.verbosity = request.verbosity;
-    	    	
-    	//ObjectMapper mapper = new ObjectMapper(); // jackson's objectmapper
-    	    	
-    	log(1, "Received /api/v1/OptimizeChargingProfiles with body: " + request.toString());
-    	
-    	// Request values
-    	Object response = null; 
-    	HttpStatus httpStatus = HttpStatus.ACCEPTED; 
-    	try {
-    		//OptimizeChargingProfilesRequest request = mapper.convertValue(payload, OptimizeChargingProfilesRequest.class);
-        	State state = request.state.toState(); 
-    		Event event = request.event.toEvent(state); 
-        	
-        	// React to events
-        	StrategyAlgorithmic strategy = new StrategyAlgorithmic(CarDepartureForecast.getDefaultCarDepartureForecast()); 
-        	strategy.objectiveEnergyCosts.setWeight(0);
-        	strategy.objectiveFairShare.setWeight(1);
-        	strategy.setReoptimizeOnStillAvailableAfterExpectedDepartureTimeslot(true);
-        	strategy.setRescheduleCarsWith0A(false);
-        	strategy.react(state, event); 
-        	
-        	
-        	// Create response
-        	response = new OptimizeChargingProfilesResponse(state.cars);
-    	}
-    	catch (Exception e) {
-    		//e.printStackTrace();
-    		System.err.println(e.getClass().getName()); 
-    		System.err.println(e.getMessage()); 
-    		response = new ErrorResponse(e); 
-    		httpStatus = HttpStatus.INTERNAL_SERVER_ERROR; 
-    		e.printStackTrace();
-    	}
-    	
-    	
-    	
-    	//String result = "End point working! Loaded " + state.cars.size() + " cars and " + state.nChargingStations + " charging stations. Charging profile produced: " + Arrays.toString(state.getCar(0).getCurrentPlan()); 
-    	return new ResponseEntity<>(response, httpStatus); 
+    public StrategyAlgorithmic buildStrategy(OptimizerSettings settings) {
+        StrategyAlgorithmic strategy = new StrategyAlgorithmic(CarDepartureForecast.getDefaultCarDepartureForecast());
+        strategy.objectiveFairShare.setWeight(settings.getWeightObjectiveFairShare());
+        strategy.objectiveEnergyCosts.setWeight(settings.getWeightObjectiveEnergyCosts());
+        strategy.objectivePeakShaving.setWeight(settings.getWeightObjectivePeakShaving());
+        strategy.objectiveLoadImbalance.setWeight(settings.getWeightObjectiveLoadImbalance());
+        strategy.setReoptimizeOnStillAvailableAfterExpectedDepartureTimeslot(true);
+        strategy.setRescheduleCarsWith0A(false);

```

Comment:
That two last have the current default I guess.

Reply:
Exactly - these two settings exist for legacy code but should not really be exposed in the REST API.

Reply:
https://github.com/SAP/emobility-smart-charging/blob/master/src/main/java/com/sap/charging/realTime/StrategyAlgorithmic.java#L40